src/weekly-test-summary-workspace/weekly-test-summary-check.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { EmissionsImportDTO } from 'src/dto/emissions.dto';
+import { WeeklyTestSummaryImportDTO } from 'src/dto/weekly-test-summary.dto';
+import { TestTypeCodes } from 'src/enums/test-type-code.enum';
+
+@Injectable()
+export class WeeklyTestSummaryCheckService {
+  constructor() {}
+
+  runChecks(payload: EmissionsImportDTO): string[] {
+    const errorList: string[] = [];
+
+    payload.weeklyTestSummaryData.forEach(wts => {
+
+        // import-38
+        let error = this.inappropriateChildrenRecordsCheck(wts);
+
+        errorList.push(error);
+    });
+
+    return errorList;
+  }
+
+  // import-38
+  private inappropriateChildrenRecordsCheck(
+    summary: WeeklyTestSummaryImportDTO,
+  ): string {
+    if (
+      summary.testTypeCode === TestTypeCodes.HGSI1 &&
+      summary?.weeklySystemIntegrityData.length > 0
```

Comment:
I think this should be:

`summary.testTypeCode !== TestTypeCodes.HGSI1`

'If the TestTypeCode is **not equal** to "HGSI1"'

"If Workspace Weekly System Integrity is found"

Reply:
I believe when the message says "you reported a record with [TestTypeCode] it want a printout of what code was given?

Reply:
Good catch on the !==. Will make that change.

And yes it wants the code printed out hence why there is the ${TestTypeCodes.HGSI1} in the message string

Reply:
I meant, the message doesn't make sense if it's !== with TestTypeCodes.HGSI1, I think it wants ${summary.testTypeCode}

Reply:
That makes sense. I'll make that change too